src/main/scala/inox/ast/Paths.scala

```diff
@@ -59,18 +63,22 @@ trait Paths { self: SymbolOps with TypeOps =>
     * not defined, whereas an encoding of let-bindings with equalities
     * could introduce non-sensical equations.
     */
-  class Path protected(val elements: Seq[Path.Element]) extends Printable with PathLike[Path] {
+  class Path protected(val elements: Seq[Path.Element], val bounds: Seq[ValDef])
```

Comment:
Not sure if this is what `Path` was designed to become...

Reply:
I think it's not a bad thing to know about bound variables in the path. It could be useful for other things than well-formedness checks at some point. However, I think it would be better if you put the bounds inside the `elements` list as well so we can preserve the order of bindings/conditions/bounds.

It's probably not a good idea to use `Either` for such things anyway, they make the code hard to read...

Reply:
I'll look into this next week